### PR TITLE
doc: Fix typos / spelling issues

### PR DESCRIPTION
**Pull request by SeanKilleen**
ℹ️ This originally started out as a request to automate spell-checking, but was reverted per request to just fix individual issues.

-------------

Hey all -- I've done this for a few repos in the past (NUnit, Marten, HotChocolate, Uno Platform). Figured I'd offer it here too.

## Work

* [x] Add cSpell GitHub Action
* [x] Add cSpell config
* [x] Address any reported cSpell issues
* [x] Separate words vs. ignoreWords in cSpell config

### CODE REVIEW

**Comment by SeanKilleen** [comment]
Down to 97 reported issues.

**Comment by SeanKilleen** [comment]
Alright -- I think this is ready for review. Only found a handful of actual misspellings; the rest was defining known terms and using back-ticks where non-word terms were used (such as around option & param names).

I'll go through now and do a review to add ℹ️ comments.

File: cSpell.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		"navigations",
+		"pluggable",
+		"Polyfilling",
+		"Pytest",
+		"sandboxing",
+		"SIGHUP",
+		"SIGTERM",
+		"stringifiable"
+	],
+	"ignoreWords": [
```
Comment (SeanKilleen) [inline cSpell.json:17]: ℹ️ Note; `words` can be suggested when spellcheck detects a misspelling in the IDE. `ignoreWords` won't be suggested.

File: cSpell.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			"name": "Snippet references 2",
+			"pattern": "\\<\\[sample:(.*)",
+			"description": "another kind of snippet reference"
+		},
+		{
+			"name": "Multi-line code blocks",
+			"pattern": "/^\\s*```[\\s\\S]*?^\\s*```/gm"
+		},
+		{
+			"name": "HTML Tags",
```
Comment (SeanKilleen) [inline cSpell.json:68]: ℹ️ I've adapted these patterns from various locations over time. `HTML Tags` is the only new one I added. It's pretty naive but got the job done in this case, catching every false positive reported issue.

File: packages/playwright-test/types/test.d.ts [hunk 3]
```diff
@@ -3625,7 +3625,7 @@ interface LocatorAssertions {
 
     /**
      * When set to `"css"`, screenshot will have a single pixel per each css pixel on the page. For high-dpi devices, this will
-     * keep screenshots small. Using `"device"` option will produce a single pixel per each device pixel, so screenhots of
+     * keep screenshots small. Using `"device"` option will produce a single pixel per each device pixel, so screenshots of
```
Comment (SeanKilleen) [inline packages/playwright-test/types/test.d.ts:3628]: ℹ️ This is the only spelling error I corrected in non-docs code. It happened to contain the same typo that was in the docs when I did a find/replace, so I figured I'd fix it here too.

File: docs/src/api/class-apirequest.md [hunk 4]
```diff
@@ -12,26 +12,26 @@ see [APIRequestContext].
 
 Creates new instances of [APIRequestContext].
 
-### option: APIRequest.newContext.useragent = %%-context-option-useragent-%%
+### option: `APIRequest.newContext.useragent = %%-context-option-useragent-%%`
```
Comment (pavelfeldman) [inline docs/src/api/class-apirequest.md:15]: Playwright generates code from the documentation, so we can't change sections in the md files.

File: cSpell.json [hunk 5]
```diff
@@ -0,0 +1,84 @@
+{
+	"version": "0.2",
+	"language": "en",
+	"words": [
```
Comment (pavelfeldman) [inline cSpell.json:4]: I don't think we can commit to maintaining a list like this. We appreciate the typo fixes, but we can't afford an include list with the dictionary words and a dependency to cspell at this point.

**Comment by SeanKilleen** [comment]
@pavelfeldman that's quite alright, I understand. Many projects don't have to maintain the list too much but especially with auto-generation it's not going to be helpful.

Let me revise the PR to just include the typos that I've found. I'll force-push the changes and update the PR title.

File: docs/src/api/class-keyboard.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 * since: v1.8
 
 Keyboard provides an api for managing a virtual keyboard. The high level api is [`method: Keyboard.type`], which takes
-raw characters and generates proper keydown, keypress/input, and keyup events on your page.
+raw characters and generates proper `keydown`, `keypress`/`input`, and `keyup` events on your page.
```
Comment (SeanKilleen) [inline docs/src/api/class-keyboard.md:5]: ❓ Please let me know if you'd like to keep or revert this change. Happy to do either.

File: docs/src/api/javascript.md [hunk 7]
```diff
@@ -9,7 +9,7 @@
 - returns: <[null]|[Object]>
   - `errorText` <[string]> Human-readable error message, e.g. `'net::ERR_FAILED'`.
 
-### param: ElementHandle.evalOnSelector.expression = %%-js-evalonselector-pagefunction-%%
+back### param: ElementHandle.evalOnSelector.expression = %%-js-evalonselector-pagefunction-%%
```
Comment (SeanKilleen) [inline docs/src/api/javascript.md:12]:
```suggestion
### param: ElementHandle.evalOnSelector.expression = %%-js-evalonselector-pagefunction-%%
```

**Comment by SeanKilleen** [comment]
I do see that the docs lint & build is failing here -- and it may be related to my changes, but I'm having difficulty understanding how. I'm happy to adjust as needed.

File: docs/src/actionability.md [hunk 8]
```diff
@@ -7,6 +7,7 @@ Playwright performs a range of actionability checks on the elements before makin
 behave as expected. It auto-waits for all the relevant checks to pass and only then performs the requested action. If the required checks do not pass within the given `timeout`, action fails with the `TimeoutError`.
 
 For example, for [`method: Page.click`], Playwright will ensure that:
+
```
Comment (mxschmitt) [inline docs/src/actionability.md:10]: revert the changes in this file?

**Comment by mxschmitt** [comment]
> I do see that the docs lint & build is failing here -- and it may be related to my changes, but I'm having difficulty understanding how. I'm happy to adjust as needed.

you need to run `npm ci && npm run build` to re-generate the generated files. This will give you the missing diff which makes CI green.

**Review by mxschmitt — APPROVED** [review]
